### PR TITLE
Closes #12 feat basic functionality

### LINKED ISSUE #12: P1, Clone / download the repo
Design and implement the functionality to Clone / download the repo into a temporary file

_Remember to document your work_

### PR DESCRIPTION
**Pull request by EleonoraBorzi**

### CODE REVIEW

**Comment by snissy** [comment]
> 
> 
> Test-comment from server :)

Nice man!

File: src/main/java/RepositoryCloner.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            File tempDirectory  = new File(repoName);
+            boolean okCloning = true;
+
+            Git git = Git.cloneRepository().
+                    setURI(repoHttpsAdress).
+                    setBranch(branchName).
+                    setDirectory(new File(tempDirectoryPath)).
+                    call();
+
+            System.out.println((okCloning ? "Successful " : "Unsuccessful ") + "cloning of repo: " + repoHttpsAdress);
```
Comment (snissy) [inline src/main/java/RepositoryCloner.java]:
```suggestion
            System.out.println("Successful cloning of repo: " + repoHttpsAdress);
```

Comment (snissy) [reply src/main/java/RepositoryCloner.java]: I wrote the previous code, with the new structure it's a bit redundant. So we can rewrite it without the one line if-statment

File: src/main/java/RepositoryCloner.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            return tempDirectoryPath;
+        }catch (Exception e){
+            System.out.println("Repo could not be cloned");
+            return null;
```
Comment (Sebberh) [inline src/main/java/RepositoryCloner.java]: Returning an empty string would be safer

**Review by Sebberh — APPROVED** [review]

File: src/main/java/RepositoryCloner.java (hunk 2 above)
Comment (EleonoraBorzi) [reply src/main/java/RepositoryCloner.java]: already fixed! Thanks!

**Review by snissy — APPROVED** [review]
LGTM, Great Work Elli!

**Pull request merged**